src/components/fields/SchemaField.js

```diff
@@ -72,17 +72,38 @@ function Label(props) {
   );
 }
 
-function LabelInput(props) {
-  const { id, label, onChange } = props;
-  return (
-    <input
-      className="form-control"
-      type="text"
-      id={id}
-      onBlur={event => onChange(event.target.value)}
-      defaultValue={label}
-    />
-  );
+class LabelInput extends React.Component {
+  constructor(props) {
+    super();
+    this.state = {
+      value: props.label,
+    };
+  }
+  render() {
+    const { id, label, onChange, schema, registry } = this.props;
+
+    if (!schema) {
+      return (
+        <input
+          className="form-control"
+          type="text"
+          id={id}
+          onBlur={event => onChange(event.target.value)}
+          defaultValue={label}
+        />
+      );
+    }
+
+    return (
+      <SchemaField
+        schema={schema}
+        formData={this.state.value}
+        registry={registry}
+        onChange={value => this.setState({ value })}
+        onBlur={(id, value) => onChange(value)}
```

Comment:
When using the dropdown to select a name for the property name, the labels for the key & value only update on the blur event -- so the user experience is kind of jarring. Is it possible to make this happen not on the on blur event, but instead on the on change event?